src/utils/network/session-helper.js

```diff
@@ -389,21 +391,99 @@ export const getUserProfileName = async (session, webId) => {
- * Gets user's name from profile using their webId
+ * A function that saves a message TTL file inside containerUrl and name it
+ * based on its slug suggestion
  *
  * @memberof utils
- * @function saveMessageTTLInInbox
+ * @function saveMessageTTL
  * @param {Session} session - Solid's Session Object (see {@link Session})
  * @param {URL} containerUrl - URL location of Pod container
  * @param {SolidDataset} solidDataset - Solid's dataset object on Pod
  * @param {string} slug - The slug suggestion for the message file
- * @returns {Promise} Promise - Fetch user's name from their Solid Pod profile
+ * @returns {Promise} Promise - Saves message TTL file inside containerUrl and
+ * name the TTL file based on its slug suggestion
  */
 
-export const saveMessageTTLInInbox = async (session, containerUrl, solidDatset, slug) => {
+export const saveMessageTTL = async (session, containerUrl, solidDatset, slug) => {
   await saveSolidDatasetInContainer(containerUrl, solidDatset, {
     slugSuggestion: `${slug}.ttl`,
     contentType: 'text/turtle',
     fetch: session.fetch
   });
 };
+
+/**
+ * A function that parses a message TTL file from inbox or outbox and returns a
+ * messageObject
+ *
+ * @memberof utils
+ * @function parseMessageTTL
+ * @param {Thing[]} messageTTLThing - List of message Things from message boxes
+ * @returns {object} messageObject - An object containinng the message content,
+ * title, uploadDate, sender, and recipient
+ */
+
+export const parseMessageTTL = (messageTTLThing) => {
+  // Get data related to #message
+  const messageThing = messageTTLThing.find((thing) => thing.url.includes('#message'));
+  const message = getStringNoLocale(messageThing, RDF_PREDICATES.message);
+  const title = getStringNoLocale(messageThing, RDF_PREDICATES.title);
+  const uploadDate = getDatetime(messageThing, RDF_PREDICATES.uploadDate);
+
+  // Get data related to #sender
+  const senderThing = messageTTLThing.find((thing) => thing.url.includes('#sender'));
+  const sender = getStringNoLocale(senderThing, RDF_PREDICATES.sender);
+
+  // Get data related to #recipient
+  const recipientThing = messageTTLThing.find((thing) => thing.url.includes('#recipient'));
+  const recipient = getStringNoLocale(recipientThing, RDF_PREDICATES.recipient);
+
+  return { message, title, uploadDate, sender, recipient };
+};
+
+/**
+ * A function that builds a new message TTL file to be sent
+ *
+ * @memberof utils
+ * @function buildMessageTTL
+ * @param {Session} session - Solid's Session Object (see {@link Session})
+ * @param {Date} date - JavaScript Date object
+ * @param {string} senderName - Name of sender
+ * @param {string} recipientName - Name of recipient
+ * @param {URL} recipientWebId - webId of recipient
+ * @param {Thing[]} messageTTLThing - List of message Things from message boxes
+ * @returns {object} messageObject - An object containinng the message content,
+ * title, uploadDate, sender, and recipient
+ */
+
+export const buildMessageTTL = (
```

Comment:
This can be in a follow up PR, but in addition to these fields, we'll probably also want:

- deliveryStatus: e.g. `pending`, `delivered`, `failed to send`, `refused`, etc.
- UUID: a value that uniquely identifies a message. Will allow us to tell if a message in one person's inbox is the same as a message in another person's outbox. Can also be used to check if a message was successfully delivered. If a message with that UUID isn't in the inbox, it wasn't delivered
- in-reply-to: a UUID of a message in the user's inbox. Allows us to implement message threading.

Reply:
Will make note of it